### ISSUE TITLE
Fan goes offline after few hours in battery mode

### ISSUE THREAD

**Issue by AlexKalopsia** [syssi/xiaomi_fan#196]
Hi, I know that this is technically not an issue with the component, but perhaps we could brainstorm a solution). 

I noticed that when the fan is off and in battery mode, after a few hours it becomes completely offline (so HASS can't find it nor communicate with it. 

`Error: Unable to discover the device 192.168.x.xxx`

[Apparently](https://www.reddit.com/r/Xiaomi/comments/o86bxp/xiaomi_smartmi_fan_3_device_is_offline/), this is a default behavior of the device, to preserve battery life. 

```
With the help of the internal battery it is possible to place the fan anywhere for use. All functionalities remain the same, but you should keep in mind that in battery mode the wireless control is switched off when the fan is turned off to save energy. This means that if the fan is turned off, it cannot be operated with the remote control or via the app. In this case, a button on the fan itself will have to be physically pressed before everything can be controlled wirelessly again.
```

This behavior makes total sense for a device that is not integrated with HASS, but it becomes very annoying otherwise. 

I am wondering if you encountered this issue before and whether there is a workaround to this issue. My thinking is that I could potentially do an " if on, turn on and then off" automation every hour or so, or perhaps pulling status information is also enough?

 My apologies if this is not the right place to discuss this.

**Comment by popy2k14** [comment]
Same issue here.
The log messages is present thousands of times in my logs in one day when the fan goes to sleep and switches off wifi.

See here:
![image](https://user-images.githubusercontent.com/6324597/185989726-bafaa25d-09de-445a-98c8-3520ae358263.png)

It's logical that the fan is switching off wifi to save power when there is no control over a few hours.
That's not the issue for me.

But flodding HA's logs with this discover message shorten the lifetime of my emmc.

@syssi Is there a way to disable those messages? Maybe with an option (device is battery powered)?

I am here to debug/help if needed.

thx

**Comment by syssi** [comment]
You can decrease the log level of the namespace to suppress errors.

**Comment by popy2k14** [comment]
thx for the hint.
Is that correct to have the default logger config (critical, fatal, error, warning) for all except the fan integration which just logs critical?

```
# logging configuration
logger:
  default: warning
  logs:
    custom_components.xiaomi_miio_fan: critical
```

**Comment by syssi** [comment]
I will have a look tomorrow and will probably change the log level of the error to warning. Without looking up the docs your configuration looks good.

**Comment by popy2k14** [comment]
That would be a nice change (error to warning) so the log issue should be solved than.
Please let me know if you done the change, i am here to test.

To come back to the OPs issue:
As mentioned, it is no big deal for me that the device is offline after some time.
Tried to send something, but sadly also the "tone/beep on/off" is making the noise.

Do you know any command which we could send in an interval wich doesn't make any noise?
If we found some command, this could be an option in an new version.

**Comment by AlexKalopsia** [comment]
I think it's making a beep because the "set tone off" automatically triggers a "turn on" message first (making the beep). I would recommend just turning off the beeping feedback, that's at least what I have done.

**Comment by AlexKalopsia** [comment]
Yes, I have an automation where every 60 minutes I send a "Turn on -> Turn off" sequence, which keeps the device alive. 
The main catch, though, is that you still end up having the same issue as before because, eventually, the battery obviously runs out and the fan becomes unreacheable until you put it on the charger.

**Comment by Sonic77** [comment]
Could the severity of this message be reduced to informational, because this is expected behavior for the S2 model.

A backoff for fetching the state could also be feasible, this message appears in the log every 30 seconds. that period could be increased and increased further after repeated failed attempts.

```
2023-06-22 01:49:53.312 ERROR (MainThread) [custom_components.xiaomi_miio_fan.fan] XiaomiFan Got exception while fetching the state: Unable to discover the device 192.168.1.64 , _retry=168
```

**Comment by popy2k14** [comment]
+1

As a workaround see my logger configuration above.
